lib/modules.nix

```diff
@@ -464,7 +464,9 @@ rec {
         disabledModules = m.disabledModules or [];
         imports = m.require or [] ++ m.imports or [];
         options = {};
-        config = addFreeformType (addMeta (removeAttrs m ["_file" "key" "disabledModules" "require" "imports" "freeformType"]));
+        config =
+          lib.throwIfNot (isAttrs m) "module ${file} (${key}) does not look like a module."
```

Comment:
I think this check should be done earlier, ideally at the start of this function, since `m` is always required to be an attribute set, not just for this specific case here.